### PR TITLE
Crash if the same wildcard source is used in multiple sources

### PR DESCRIPTION
**Pull request by HofiOne**
Fixes: #5090

Signed-off-by: Hofi <hofione@gmail.com>

### CODE REVIEW

**Comment by kira-syslogng** [comment]
Build FAILURE

**Comment by kira-syslogng** [comment]
Build FAILURE

**Comment by kira-syslogng** [comment]
Build FAILURE

File: lib/cfg-tree.c [hunk 1]
```diff
@@ -645,6 +646,59 @@ log_expr_node_lookup_flag(const gchar *flag)
...
+}
+
+static void
+cfg_tree_stop_registering_pipes_with_persist_name(CfgTree *self)
+{
+  if (self->pipes_with_persis_name)
+    {
+      main_loop_assert_main_thread();
+
+      g_hash_table_unref(self->pipes_with_persis_name);
```
Comment (kovgeri01) [inline lib/cfg-tree.c:670]: Iterate over and unref the logpipes inside just in case something forgots to do it before this call

Comment (HofiOne) [reply lib/cfg-tree.c:670]: it is automatic if you use g_hash_table_new_full

https://docs.gtk.org/glib/type_func.HashTable.unref.html
https://docs.gtk.org/glib/type_func.HashTable.new_full.html

**Review by kovgeri01 — APPROVED** [review]

**Pull request merged**